src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/NavigationBar.tsx

```diff
@@ -0,0 +1,217 @@
+  li2: {
+    float: 'left', 
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    backgroundColor: "#022f51",
+    [theme.breakpoints.down(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  li2Mounted: {
+    marginBottom: '6px',
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    backgroundColor: "#022f51",
+    [theme.breakpoints.up(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  li3: {
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    backgroundColor: "#022f51",
+    [theme.breakpoints.up(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  a: {
+    display: 'block', 
+    textAlign: "center", 
+    padding: '12px', 
+    borderBottom: '0',
+    "&:hover": {
+      borderBottom: "0px solid rgba(0,0,0,0)"
+    }
+  },
+  a2: {
+    display: 'block',   
+    textAlign: "center", 
+    padding: '12px', 
+    borderBottom: '0',
+    color: 'white!important',
+    "&:hover": {
+      borderBottom: "0px solid rgba(0,0,0,0)"
+    },
+    width: '100%',
+    fontSize: '1.5rem'
+  },
+  i: {
+    color: 'white'
+  },
+  hide: {
+    display: 'none'
+  }
+});
+
+export interface INavigationBar {
+  id: string;
+  showBackButton: boolean;
+  textResourceBindings: any;
+  triggers?: Triggers[];
+}
+
+export function NavigationBar(props: INavigationBar) {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const orderedLayoutKeys = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.layoutOrder);
+  const returnToView = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.returnToView);
+  const pageTriggers = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.pageTriggers);
+  const triggers = props.triggers || pageTriggers;
+  const textResources = useSelector((state: IRuntimeState) => state.textResources.resources);
+  const currentView: string = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.currentView);
+
+
+
+  const NavBar = () => {
+
+    const [mounted, setMounted] = useState(false);
+
+    useEffect(() => {
+      console.log("Mounted")
+    }, [])
+
+    const OnClickNav = (index: string) => {
+
+      const runPageValidations = !returnToView && triggers?.includes(Triggers.ValidatePage);
+      const runAllValidations = returnToView || triggers?.includes(Triggers.ValidateAllPages);
+      const runValidations = (runAllValidations && 'allPages') || (runPageValidations && 'page') || null;
+
+      dispatch(FormLayoutActions.updateCurrentView({ newView: index, runValidations }));
+
+    };
+
+    const pageList = orderedLayoutKeys.map((view) => 
+      <li className={ currentView == view ? classes.li2 : classes.li } >
```

Comment:
If you regard the comment I wrote earlier you can replace ``view``  with ``item``
There are a lot of components here that could be split into separate files and reduce this file's overall complexity.
Also the fact that classes have names similar to the markup is not very helpful for the reader.

Reply:
Do you have any suggestion for filenames to split into? I see the issue, but is there some best practice for solving it?

Reply:
What I usually do is to replace the main component file with a directory with the same name and copy all it's content into an `index.ts` file to begin with.
If done correctly all external references to the file should compile as before.
Then inside the new directory, I create new files containing chunks that I extract from the `index.ts` file one by one.
Eventually your index.ts file contains only some imports, the shell of the original component, and a few lines of implementation of the extracted [smaller] components.